### PR TITLE
[@types/relay-runtime]: 10.1.0 - CacheConfig

### PR DESCRIPTION
**Pull request by morrys**
In this PR I fix:
[RequestDescriptor](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/store/RelayStoreTypes.js#L83-L88)
[createOperationDescriptor](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/store/RelayModernOperationDescriptor.js#L35-L40)
[createRequestDescriptor](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/store/RelayModernOperationDescriptor.js#L70-L74)
[fetchQuery internal](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/query/fetchQueryInternal.js#L105-L108)

```js
export type RequestDescriptor = {|
  +identifier: RequestIdentifier,
  +node: ConcreteRequest,
  +variables: Variables,
  +cacheConfig: ?CacheConfig,
|};

function createOperationDescriptor(
  request: ConcreteRequest,
  variables: Variables,
  cacheConfig?: ?CacheConfig,
  dataID?: DataID = ROOT_ID,
): OperationDescriptor

function createRequestDescriptor(
  request: ConcreteRequest,
  variables: Variables,
  cacheConfig?: ?CacheConfig,
): RequestDescriptor

function fetchQuery(
  environment: IEnvironment,
  operation: OperationDescriptor,
): Observable<GraphQLResponse>

```

* RelayEnvironment

[execute](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/store/RelayModernEnvironment.js#L362-L368)
[executeMutation](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/store/RelayModernEnvironment.js#L413-L425)
```js
  execute({
    operation,
    updater,
  }: {|
    operation: OperationDescriptor,
    updater?: ?SelectorStoreUpdater,
  |}): RelayObservable<GraphQLResponse>

 executeMutation({
    operation,
    optimisticResponse,
    optimisticUpdater,
    updater,
    uploadables,
  }: {|
    operation: OperationDescriptor,
    optimisticUpdater?: ?SelectorStoreUpdater,
    optimisticResponse?: ?Object,
    updater?: ?SelectorStoreUpdater,
    uploadables?: ?UploadableMap,
  |}): RelayObservable<GraphQLResponse>
```

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
[mutationConfig](https://github.com/facebook/relay/blob/v10.0.1/packages/relay-runtime/mutations/commitMutation.js#L62-L84)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by sibelius — APPROVED** [review]
nice

File: types/relay-runtime/lib/store/RelayModernOperationDescriptor.d.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
  * Creates an instance of the `OperationDescriptor` type defined in
  * `RelayStoreTypes` given an operation and some variables. The input variables
  * are filtered to exclude variables that do not match defined arguments on the
  * operation, and default values are populated for null values.
  */
-export function createOperationDescriptor(request: ConcreteRequest, variables: Variables): OperationDescriptor;
+export function createOperationDescriptor(
+    request: ConcreteRequest,
+    variables: Variables,
+    cacheConfig?: CacheConfig | null | undefined,
```
Comment (maraisr) [inline types/relay-runtime/lib/store/RelayModernOperationDescriptor.d.ts:13]: Think the `| undefined` isn't needed

File: types/relay-runtime/lib/store/RelayModernOperationDescriptor.d.ts [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+    dataID?: DataID,
+): OperationDescriptor;
 
-export function createRequestDescriptor(request: ConcreteRequest, variables: Variables): RequestDescriptor;
+export function createRequestDescriptor(
+    request: ConcreteRequest,
+    variables: Variables,
+    cacheConfig?: CacheConfig | null | undefined,
```
Comment (maraisr) [inline types/relay-runtime/lib/store/RelayModernOperationDescriptor.d.ts:20]: Same here

File: types/relay-runtime/lib/store/RelayStoreTypes.d.ts [hunk 3]
```diff
@@ -63,6 +63,7 @@ export interface RequestDescriptor {
     readonly identifier: RequestIdentifier;
     readonly node: ConcreteRequest;
     readonly variables: Variables;
+    readonly cacheConfig: CacheConfig | null | undefined;
```
Comment (maraisr) [inline types/relay-runtime/lib/store/RelayStoreTypes.d.ts:66]: And here 😅

**Review by maraisr — CHANGES_REQUESTED** [review]
Just a few trivial things. On the outset thanks so much for doing this! ❤️ 

Think we can get some tests in there as well for this?

**Comment by maraisr** [comment]
> Just a few trivial things. On the outset thanks so much for doing this! ❤️ 
> 
> Think we can get some tests in there as well for this? 

I could help with this too ☝🏻

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. ReplaySubject
1. MutationTypes
1. RangeOperations
1. MutationHandlers
1. VIEWER_ID

<details>
<summary>as well as these 9 other properties...</summary>
<p>VIEWER_TYPE, PreloadableQueryRegistry, createPayloadFor3DField, RelayError, DEFAULT_HANDLE_KEY, isScalarAndEqual, recycleNodesInto, stableCopy, getFragmentIdentifier</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5dde6ea5463485e7f1eb4c9aad98a76d49398f75
</p>

**Review by alloy — COMMENTED** [review]
I agree that it would be nice to have a few snippets of test code, at least for the operation descriptor functions.

Also, this seems like a great time to add yourself back as maintainer of the types. If you’re still interested in being kept up-to-date, mind adding yourself back?

**Comment by alloy** [comment]
Thanks!

**Pull request merged**